src/features/nameChecker/checkNameAgainstPatterns.ts

```diff
@@ -8,18 +10,20 @@ type Pattern = RuntimeConfigurationSchema['nameChecker']['patterns'][number]
 export function checkNameAgainstPatterns(
   name: string,
   patterns: Pattern[],
+  isZalgoEnabled: boolean,
   log: Logger = console,
-): RegExp | undefined {
+): boolean | undefined {
+  if (isZalgoEnabled && isZalgo(name.trim())) {
+    return true
+  }
   for (const pattern of patterns) {
     try {
       let regexp = compiled.get(pattern.regexp)
       if (!regexp) {
         regexp = new RegExp(pattern.regexp, 'i')
         compiled.set(pattern.regexp, regexp)
       }
-      if (regexp.test(name) || regexp.test(name.replaceAll(/\s+/g, ''))) {
-        return regexp
-      }
+      return regexp.test(name) || regexp.test(name.replaceAll(/\s+/g, ''))
```

Comment:
If there are multiple patterns, and the 1st pattern doesn't match, this will cause the function to return false without considering he rest of the patterns.

Did you mean:

```js
if (regexp.test(name) || regexp.test(name.replaceAll(/\s+/g, ''))) {
  return true
}

// after the loop
return false
```